### ISSUE TITLE
[Enhancement]: Prompt to reassign API keys when removing a user

### ISSUE THREAD

**Issue by odirlon** [woothemes/woocommerce#49342]
### Describe the solution you'd like

When removing a user, Wordpress removes the rest APIs created by that user. Even when we assign that user's content that is being removed to another Administrator. We believe this part should also be attributed.
The list of users could also be reformulated to make selection in stores with many users viable. Select is limited by the machine being used, a store with 100 thousand users makes it impossible to display the select, imagine in stores with 500 thousand, 800 thousand...

### Describe alternatives you've considered

API - We are keeping users still active with a private email and a more difficult password until we recreate the APIs.
User selection - We select to assign a user, remove the field from the select via "Inspect Element", create it with just one Option (Already with the ID of the user who will receive the information) and finish the operation. However, if it were in ajax it would be solved.

**Comment by David-Marchiori** [comment]
I agree with your observation; this happens to me as well. I end up doing double the work because I have to recreate the keys and reintegrate everything..

**Comment by David-Marchiori** [comment]
I agree with your observation; this happens to me as well. I end up doing double the work because I have to recreate the keys and reintegrate everything. I suggest this be reviewed.

**Comment by coreymckrill** [comment]
Hi @odirlon, could you clarify what you mean here? Are you referring to the API keys you can create in WooCommerce settings?

**Comment by odirlon** [comment]
Yes, the woocommer settings api keys, sorry. 

 That's not so much the problem.  I think the biggest month
 The issue would be that it is not possible to choose the right one when removing users because for a store with a lot of sales the page gets stuck.

**Comment by nerrad** [comment]
Duplicate of #58357. Although this was created earlier, I didn't notice this until after leaving a comment on the other issue. My comment there applies to this as well.
